File: trunk/cpp/DIS/RadioCommunicationsPdu.cpp
#include <DIS/RadioCommunicationsPdu.h> 

using namespace DIS;


RadioCommunicationsPdu::RadioCommunicationsPdu() : Pdu(),
   _entityId(), 
   _radioId(0)
{
    setProtocolFamily( 4 );
}

RadioCommunicationsPdu::~RadioCommunicationsPdu()
{
}

EntityID& RadioCommunicationsPdu::getEntityId() 
{
    return _entityId;
}

const EntityID& RadioCommunicationsPdu::getEntityId() const
{
    return _entityId;
}

void RadioCommunicationsPdu::setEntityId(const EntityID &pX)
{
    _entityId = pX;
}

unsigned short RadioCommunicationsPdu::getRadioId() const
{
    return _radioId;
}

void RadioCommunicationsPdu::setRadioId(unsigned short pX)
{
    _radioId = pX;
}

void RadioCommunicationsPdu::marshal(DataStream& dataStream) const
{
    Pdu::marshal(dataStream); // Marshal information in superclass first
    _entityId.marshal(dataStream);
    dataStream << _radioId;
}

void RadioCommunicationsPdu::unmarshal(DataStream& dataStream)
{
    Pdu::unmarshal(dataStream); // unmarshal information in superclass first
    _entityId.unmarshal(dataStream);
    dataStream >> _radioId;
}


bool RadioCommunicationsPdu::operator ==(const RadioCommunicationsPdu& rhs) const
 {
     bool ivarsEqual = true;

     ivarsEqual = Pdu::operator==(rhs);

     if( ! (_entityId == rhs._entityId) ) ivarsEqual = false;
     if( ! (_radioId == rhs._radioId) ) ivarsEqual = false;

    return ivarsEqual;
 }

int RadioCommunicationsPdu::getMarshalledSize() const
{
   int marshalSize = 0;

   marshalSize = Pdu::getMarshalledSize();
   marshalSize = marshalSize + _entityId.getMarshalledSize();  // _entityId
   marshalSize = marshalSize + 2;  // _radioId
    return marshalSize;
}

File: trunk/java/edu/nps/moves/dis/RadioCommunicationsPdu.java
package edu.nps.moves.dis;

import java.util.*;
import java.io.*;

/**
 * Section 5.3.8. Abstract superclass for radio communications PDUs.
 *
 * Copyright (c) 2007, MOVES Institute, Naval Postgraduate School. All rights reserved.
 * This work is licensed under the BSD open source license, available at https://www.movesinstitute.org/licenses/bsd.html
 *
 * @author DMcG
 */
public class RadioCommunicationsPdu extends Pdu
{
   /** ID of the entitythat is the source of the communication */
   protected EntityID  entityId = new EntityID(); 

   /** particular radio within an entity */
   protected int  radioId;


/** Constructor */
 public RadioCommunicationsPdu()
 {
    setProtocolFamily( (short)4 );
 }

public int getMarshalledSize()
{
   int marshalSize = 0; 

   marshalSize = super.getMarshalledSize();
   marshalSize = marshalSize + entityId.getMarshalledSize();  // entityId
   marshalSize = marshalSize + 2;  // radioId

   return marshalSize;
}


public void setEntityId(EntityID pEntityId)
{ entityId = pEntityId;
}

public EntityID getEntityId()
{ return entityId; }

public void setRadioId(int pRadioId)
{ radioId = pRadioId;
}

public int getRadioId()
{ return radioId; 
}


public void marshal(DataOutputStream dos)
{
    super.marshal(dos);
    try 
    {
       entityId.marshal(dos);
       dos.writeShort( (short)radioId);
    } // end try 
    catch(Exception e)
    { 
      System.out.println(e);}
    } // end of marshal method

public void unmarshal(DataInputStream dis)
{
    super.unmarshal(dis);

    try 
    {
       entityId.unmarshal(dis);
       radioId = dis.readShort();
    } // end try 
   catch(Exception e)
    { 
      System.out.println(e); 
    }
 } // end of unmarshal method 


 /**
  * The equals method doesn't always work--mostly on on classes that consist only of primitives. Be careful.
  */
 public boolean equals(RadioCommunicationsPdu rhs)
 {
     boolean ivarsEqual = true;

    if(rhs.getClass() != this.getClass())
        return false;

     if( ! (entityId.equals( rhs.entityId) )) ivarsEqual = false;
     if( ! (radioId == rhs.radioId)) ivarsEqual = false;

    return ivarsEqual;
 }
} // end of class
